### DISCUSSION TITLE
How could I apply a number of controller actions in a single frame?

### DISCUSSION THREAD

**Discussion by Preston-Harrison** [idanarye/bevy-tnua#72] [Q&A]
I'm considering switching to tnua from the default rapier character controller as the tnua one is much better. The only thing is my game is multiplayer with client prediction, so I'm not sure how I can run a single iteration of prediction for this controller (with either rapier or avian). My current best guess would be to move the tnua system set [`TnuaUserControlsSystemSet`](https://github.com/idanarye/bevy-tnua/blob/main/src/lib.rs#L154) to a custom schedule, then when I need to predict, I would run this schedule multiple times with [`world.run_schedule`](https://docs.rs/bevy/latest/bevy/ecs/world/struct.World.html#method.run_schedule) while applying each controller bases and actions in between schedule runs.

I would really appreciate some guidance here. Does my approach make sense, or am I over/under complicating things?

**Comment by idanarye** [comment]
Never did this prediction thing, so I don't know how it should work myself. But I assume that other than moving the system set to a different schedule, you'll also need to be able to clone the `TnuaController` component (so that you could run multiple things from the same start position)?
